apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx

```diff
@@ -34,3 +37,41 @@ export function AccountMultiSelect({
 		</ToggleGroup.Root>
 	);
 }
+
+export function AccountMultiSelectWithControls({
+	selectedAccountIDs: selectedAccountsFromProps,
+	accounts,
+	onChange: onChangeFromProps,
+}: AccountMultiSelectProps) {
+	const [selectedAccounts, setSelectedAccounts] = useState(selectedAccountsFromProps);
+	const onChange = (value: string[]) => {
+		setSelectedAccounts(value);
+		onChangeFromProps(value);
+	};
+	return (
+		<div className="flex flex-col gap-3 [&>button]:border-none">
+			<AccountMultiSelect
+				selectedAccountIDs={selectedAccounts}
+				accounts={accounts}
+				onChange={onChange}
+			/>
+
+			<Button
+				onClick={() => {
+					if (selectedAccounts.length < accounts.length) {
+						onChange(accounts.map((account) => account.address));
```

Comment:
Is this suppose to be the address or account id? There are cases that multiple accounts have the same address (rare but possible) but the id would always be unique.

Reply:
updated to use account ID